DevTools/Compiler/C++Compiler/LibCpp/Option.cpp

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2020, Denis Campredon <deni_@hotmail.fr>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "Option.h"
+#include <string.h>
+
+Cpp::Option Cpp::Option::parse_options(int ac, const char** av)
+{
+    assert(ac == 2 || ac == 4);
+
+    Cpp::Option options = {};
+
+    options.input_file = av[1];
+    if (ac == 2) {
+        options.output_file = options.input_file.isolated_copy();
+        assert(options.output_file.replace(".cpp", ".s") == 1);
```

Comment:
I'm not sure what's going on with this `replace()` method, but it presumably modifies something?
Please don't put that sort of thing inside `assert()` - that will not be run when assertions are not enabled.
also, I think it's a little bit more idiomatic to use `ASSERT()`, but meh.

Reply:
The assert have been removed. Maybe I should add it but in another line.

Reply:
There's nothing wrong with asserting, in fact, do more of that when you're assuming things!
just don't put state-changing code inside it is all I'm saying